stdlib/public/Concurrency/Task.swift

```diff
@@ -194,18 +194,29 @@ extension Task {
     }
   }
 
-  /// Indicates that the task should stop running.
+  /// Marks this task as cancelled.
   ///
   /// Task cancellation is cooperative:
   /// a task that supports cancellation
-  /// checks whether it has been canceled at various points during its work.
+  /// checks whether it has been canceled at various points during its work,
+  /// and may either return early or throw an error in order to
+  /// finish running earlier than it would if it wasn't cancelled.
```

Comment:
Suggestion:

> Cancels this task.
>
> Cancelling a task has three primary effects:
> - It flags the task as cancelled.
> - It causes any active cancellation handlers on the task to run.
> - It cancels any child tasks and task groups of the task, including
>   those created in the future.
>
> Task cancellation is cooperative in the sense that cancelling a
> task does not automatically cause arbitrary functions on the task
> to stop running or throw errors. A function may choose to react
> to cancellation by ending its work early, and it is conventional to
> signal that to callers by throwing `CancellationError`. However,
> a function that doesn't specifically check for cancellation will
> run to completion normally even if the task it is running on is
> cancelled. (Of course, it may still end early if it calls something
> else that handles cancellation by throwing and then doesn't
> handle the error.)
>
> It is safe to cancel a task from any task or thread. It is safe for
> multiple tasks or threads to cancel the same task at the same
> time. Cancelling a task that has already been cancelled has no
> additional effect.
>
> `cancel` may need to acquire locks and synchronously run
> arbitrary cancellation-handler code associated with the
> cancelled task. To reduce the risk of deadlock, it is
> recommended that callers release any locks they might be
> holding before they call `cancel`.
>
> - SeeAlso: `Task.checkCancellation()`
> - SeeAlso: `withTaskCancellationHandler()`

Reply:
That's pretty comprehensive, lemme take that -- thanks @rjmccall